apps/scan/backend/src/interpret.ts

```diff
@@ -182,7 +183,25 @@ async function nhInterpret(
     return result;
   }
 
-  const [frontResult, backResult] = result.ok();
+  const [frontResult, backResult] = mapSheet(result.ok(), (pageResult) => {
+    if (config.precinctSelection.kind !== 'AllPrecincts') {
+      if (
+        'metadata' in pageResult.interpretation &&
+        pageResult.interpretation.metadata.precinctId !==
+          config.precinctSelection.precinctId
+      ) {
+        return {
+          ...pageResult,
+          interpretation: typedAs<PageInterpretation>({
+            type: 'InvalidPrecinctPage',
+            metadata: pageResult.interpretation.metadata,
+          }),
+        };
+      }
+    }
+
+    return pageResult;
+  });
```

Comment:
Why did this need to be added in here? Were we not checking anymore?

Reply:
This actually fixes an outstanding bug where we were expecting the interpreters to do this check, but the NH interpreter never did.